servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/InstrumentationHelper.java

```diff
@@ -40,4 +47,25 @@ abstract Single<StreamingHttpResponse> trackRequest(
             Function<StreamingHttpRequest, Single<StreamingHttpResponse>> requestHandler,
             RequestInfo requestInfo,
             Context parentContext);
+
+    static SpanNameExtractor<RequestInfo> spanNameExtractor(OpenTelemetryFilterBuilder<?> builder,
+                                                                      SpanNameExtractor<RequestInfo> defaultExtractor) {
+        @Nullable Function<HttpRequestMetaData, String> customExtractor = builder.spanNameExtractor;
+        if (customExtractor == null) {
+            return defaultExtractor;
+        }
+        return requestInfo -> {
+            String result;
+            try {
+                result = customExtractor.apply(requestInfo.request());
+            } catch (Exception ex) {
+                LOGGER.warn("Failed to extract custom span name", ex);
+                result = "<error extracting name>";
+            }
+            if (result == null) {
+                result = defaultExtractor.extract(requestInfo);
```

Comment:
that's interesting. Why would we fall back to the default if an explicit extractor is set? Should we define our own function that has a Non-Null response explicitly to prevent that? Alternatively, at least we should document this in the javadoc for the custom extractor on the builder, wdyt?